### PR TITLE
Remove ad-hoc parser in wcon

### LINKED ISSUE #243: [wcon] Remove opm workaround, UDA should be available in opm now.
https://github.com/equinor/ecl2df/blob/d1df31312471e04b143798fc3b23b7b492f745d8/ecl2df/wcon.py#L43

### PR DESCRIPTION
**Pull request by berland**
Also make testing more explicit.

Resolves #243

### CODE REVIEW

File: tests/test_wcon.py [hunk 1]
```diff
@@ -69,43 +25,152 @@ def test_wcon2df():
...
+    wconhist_df = wcon.df(deck)
+    pd.testing.assert_frame_equal(
+        wconhist_df,
+        pd.DataFrame(
+            [
+                {
+                    "WELL": "FOO",
+                    "STATUS": "0",
+                    "CMODE": "1",
+                    "ORAT": 0,
```
Comment (tklov11) [inline tests/test_wcon.py:45]: Er 0 og None det samme her? Ser fra WSEGVALV testen med max blank at jeg tester mot None og ikke 0 for elementer som ikke er satt i tekststrengen.

File: tests/test_wcon.py [hunk 2]
```diff
@@ -69,43 +25,152 @@ def test_wcon2df():
...
 
-def test_str2df():
-    """Test dataframe extraction from strings"""
+def test_wconhist():
+    """Test WCONHIST parsing and column names"""
     wconstr = """
 WCONHIST
   'FOO' 0 1 /
  /
 """
```
Comment (tklov11) [inline tests/test_wcon.py:34]: Blir det penere kode når teksten i wconstr har et ekstra innrykk?

File: tests/test_wcon.py [hunk 3]
```diff
@@ -69,43 +25,152 @@ def test_wcon2df():
...
+    )
 
+
+def test_wconinjh():
+    """Test WCONINJH parsing and column names"""
     wconstr = """
 WCONINJH
   'FOO' 0 1 /
  /
 """
```
Comment (tklov11) [inline tests/test_wcon.py:67]: Samme spml som tekststrengen over

File: tests/test_wcon.py (hunk 1 above)
Comment (berland) [reply tests/test_wcon.py:45]: `0` and `None` are different Python objects, but sometimes act similarly. The statement

```
if not something:
    do_something()
```
will f.ex treat `something=0` and `something=None` identically.

In the dataframe, `0` and `None`means different things, and should be handled differently.

File: ecl2df/wcon.py [hunk 4]
```diff
@@ -1,18 +1,16 @@
-#!/usr/bin/env python
```
Comment (tklov11) [inline ecl2df/wcon.py:1]: Er denne linja nødvendig?

File: tests/test_wcon.py (hunk 2 above)
Comment (berland) [reply tests/test_wcon.py:34]: Perhaps,  but that is also a different Eclipse deck with whitespace preceding the Eclipse keyword at line start (which is not common in Eclipse files)

File: ecl2df/wcon.py (hunk 4 above)
Comment (berland) [reply ecl2df/wcon.py:1]: No, it should be removed!

**Review by tklov11 — APPROVED** [review]
LGTM

**Pull request merged**